api/src/main/java/org/openmrs/module/reporting/serializer/ReportingSerializer.java

```diff
@@ -115,4 +122,22 @@ public void serializeToStream(Object object, OutputStream out) {
             throw new IllegalStateException("Unsupported encoding", e);
         }
     }
+
+	private void setupXStreamSecurity(XStream xstream) throws SerializationException {
+		try {
```

Comment:
I feel that instead of checking for the running version of openmrs, it is this method implementation that needs to be fixed.

Reply:
@dkayiwa there was an issue on the core that caused it not to throw the API exception. I made a fix for that: https://github.com/openmrs/openmrs-core/pull/4843